### PR TITLE
Admin send mail notice request

### PR DESCRIPTION
**Pull request by maiq721**

### CODE REVIEW

**Comment by maiq721** [comment]
ready
Link task : https://dev.framgia.com/issues/184232

File: app/assets/javascripts/list_order.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+function change2(){
+	$('.unchecked-list').hide()
+	$('.checked-list').show()
+}
+function change1(){
+	$('.unchecked-list').show()
+	$('.checked-list').hide()
```
Comment (mdmahmudurrahman) [inline app/assets/javascripts/list_order.js]: 1. 2 space indentation (dap2)
2. Please the code snippet inside the following 

```ruby
$(document).ready(function() {
   .............
   ...........
});
```

File: app/controllers/admin/orders_controller.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+		@order_checked = Order.list_checked
+	end
+
+	def update
+		@order = Order.find_by id: params[:id]
+    @order.update_attributes status: params[:status]
+    Admin::OrderMailer.notice_request(@order).deliver_now
+    flash[:info] = t ".sent_email"
+    redirect_to admin_orders_path
+	end
```
Comment (mdmahmudurrahman) [inline app/controllers/admin/orders_controller.rb]: Indentation

File: app/mailers/admin/order_mailer.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class Admin::OrderMailer < ApplicationMailer
+
```
Comment (mdmahmudurrahman) [inline app/mailers/admin/order_mailer.rb]: Remove this empty line

File: app/mailers/admin/order_mailer.rb [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+  def notice_request order
+    @order = order
+
```
Comment (mdmahmudurrahman) [inline app/mailers/admin/order_mailer.rb]: The same here

File: app/models/order.rb [hunk 5]
```diff
@@ -10,5 +10,7 @@ class Order < ApplicationRecord
   validates :address, presence: true
 
   scope :by_id, -> user{where user_id: user}
+  scope :list_unchecked, ->{where status: 0}
+  scope :list_checked, ->{where status: 1}
```
Comment (mdmahmudurrahman) [inline app/models/order.rb]: Combine 2 scopes into one
```ruby
scope :by_status, ->(value){where status: value}
```

File: app/views/admin/order_mailer/notice_request.text.erb [hunk 6]
```diff
@@ -0,0 +1,5 @@
+<p><%= t ".hi" %> <%= @order.user.username %>
+</p>
```
Comment (mdmahmudurrahman) [inline app/views/admin/order_mailer/notice_request.text.erb]:
```ruby 
<p>
  <%= t ".hi" %> <%= @order.user.username %>
</p>
or 
<p><%= t ".hi" %> <%= @order.user.username %></p>
```

File: app/views/admin/order_mailer/notice_request.text.erb [hunk 7]
```diff
@@ -0,0 +1,5 @@
...
+
+<p><%= t ".invite" %></p>
+<%= link_to t("."), root_url %>
```
Comment (mdmahmudurrahman) [inline app/views/admin/order_mailer/notice_request.text.erb]: Just only a dot(.) as a key! (nerd)

File: app/views/admin/orders/index.html.erb [hunk 8]
```diff
@@ -0,0 +1,77 @@
+<% provide :title, t(".title") %>
+<h2><%= t ".title" %></h2>
+<div class="chose">
+  <button id="chose-unchecked" onclick="change1()" class="btn-primary btn"><%= t ".uncheck-list" %></button>
+  <button id="chose-checked" onclick="change2()" class="btn-primary btn"><%= t ".checked-list" %></button>
```
Comment (mdmahmudurrahman) [inline app/views/admin/orders/index.html.erb]: Add the event handler in the separate JS file using id

File: app/views/admin/orders/index.html.erb [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+            </thead>
+            <tbody>
+              <% @order_unchecked.each do |order| %>
+                <tr>
+                  <td><%= order.id %></td>
+                  <td><%= order.user.username %></td>
+                  <td><%= order.user.email %></td>
+                  <td><%= order.created_at %></td>
+                  <td><%= order.total %></td>
+                  <td><%= link_to t(".check"), admin_order_path(order.id, params.permit().merge(status: :checked)), method: :put, class: "btn btn-primary" %></td>
```
Comment (mdmahmudurrahman) [inline app/views/admin/orders/index.html.erb]: 1. `params.permit().merge(status: :checked)`
  Do this operation in the controller. 
2. Break a long line

File: app/views/layouts/admin/mailer.html.erb [hunk 10]
```diff
@@ -0,0 +1,13 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
+    <style>
+      /* Email styles need to be inline */
```
Comment (mdmahmudurrahman) [inline app/views/layouts/admin/mailer.html.erb]: Please remove comment

File: config/routes.rb [hunk 11]
```diff
@@ -10,6 +10,8 @@
 
   namespace :admin do
     resources :authors, except: :destroy
+    resources :books
+    resources :orders
```
Comment (mdmahmudurrahman) [inline config/routes.rb]: Specify the actions.

**Comment by mdmahmudurrahman** [comment]
commented

**Comment by maiq721** [comment]
ready

File: app/assets/javascripts/list_order.js [hunk 12]
```diff
@@ -0,0 +1,10 @@
+$(document).ready(function(){
+  $('#chose-unchecked').click(function(){
+  	$('.unchecked-list').show();
+	$('.checked-list').hide()
+  });
+  $('#chose-checked').click(function(){
+  	$('.checked-list').show();
+	$('.unchecked-list').hide()
```
Comment (mdmahmudurrahman) [inline app/assets/javascripts/list_order.js]: 2 space indentation
```ruby
$(document).ready(function(){
  $('#chose-unchecked').click(function(){
    $('.unchecked-list').show();
    $('.checked-list').hide()
  });
  $('#chose-checked').click(function(){
    $('.checked-list').show();
    $('.unchecked-list').hide()
  });
});
```

File: app/controllers/admin/orders_controller.rb [hunk 13]
```diff
@@ -0,0 +1,14 @@
+class Admin::OrdersController < ApplicationController
+  def index
+    @order_unchecked = Order.by_status 0
+    @order_checked = Order.by_status 1
```
Comment (mdmahmudurrahman) [inline app/controllers/admin/orders_controller.rb]: Place the constants in the settings.yml

File: app/controllers/admin/orders_controller.rb [hunk 14]
```diff
@@ -0,0 +1,14 @@
...
+  end
+
+  def update
+    @order = Order.find_by id: params[:id]
```
Comment (mdmahmudurrahman) [inline app/controllers/admin/orders_controller.rb]: 1. Add a before_action filter to load the order and check nil

File: app/models/order.rb [hunk 15]
```diff
@@ -10,5 +10,6 @@ class Order < ApplicationRecord
   validates :address, presence: true
 
   scope :by_id, -> user{where user_id: user}
+  scope :by_status, -> value{where status: value}
```
Comment (mdmahmudurrahman) [inline app/models/order.rb]:
```ruby
scope :by_status, ->value{where status: value}
```

File: app/views/admin/order_mailer/notice_request.text.erb [hunk 16]
```diff
@@ -0,0 +1,6 @@
+<p>
+	<%= t ".hi" %> <%= @order.user.username %>
```
Comment (mdmahmudurrahman) [inline app/views/admin/order_mailer/notice_request.text.erb]: Indentation

**Comment by mdmahmudurrahman** [comment]
commented

**Comment by mdmahmudurrahman** [comment]
commented
This (pr) can be (merged)
CC: [To:1818111] Nguyen Thi Huong (94)
CC: [To:2076564] Nguyen Huy Hung C (93)

**Pull request merged**